packages/sitecore-jss/src/dictionary-service.ts

```diff
@@ -0,0 +1,85 @@
+import { AxiosResponse } from 'axios';
+import mcache from 'memory-cache';
+import { AxiosDataFetcher } from './data-fetcher';
+import { DictionaryPhrases, DictionaryServiceData } from './dataModels';
+import { HttpJsonFetcher } from './httpClientInterface';
+
+export type DictionaryServiceConfig = {
+  apiHost: string;
+  apiKey: string;
+  jssAppName: string;
+  /**
+   * Custom data fetcher
+   * @see HttpJsonFetcher<T>
+   */
+  dataFetcher?: HttpJsonFetcher<DictionaryServiceData>;
```

Comment:
Do we want to follow the `dataFetcherResolver` pattern like layout-service for consistency?

Reply:
I'm not sure that we need it, I remember that we used this approach, because we had a different `return` types for dataFetcher: `layoutServiceData`, `placeholderData`. And now we have only `DictionaryPhrases`. Are you agree?

Reply:
@sc-illiakovalenko I think it was more about needing to pass along the req and res parameters (i.e. for header access/manipulation). Not sure we'd ever need to do that for dictionary, but I'm just thinking about consistency in our APIs.

Reply:
@ambrauer Agree, but seems like we didn't make manipulations with headers when request dictionary

Reply:
@sc-illiakovalenko Correct, we don't - and the req/res parameters are optional with the `DataFetcherResolver` anyway. Guess I'm just thinking we could use the same config mechanism. i.e.:
1. move `DataFetcherResolver` from `layout-service.ts` to `data-fetcher.ts`
2. switch dictionary-service to use same `dataFetcherResolver?: DataFetcherResolver;` for config